src/main/java/net/minestom/server/entity/Player.java

```diff
@@ -2244,62 +2220,62 @@ protected void showPlayer(@NotNull PlayerConnection connection) {
 
     @Override
     public @NotNull ItemStack getItemInMainHand() {
-        return inventory.getItemInMainHand();
+        return inventory.getEquipment(EquipmentSlot.MAIN_HAND, getHeldSlot());
```

Comment:
I am unclear on the need for `getEquipment` at all

Reply:
I found it useful to have a function that takes the enum instance, since otherwise every usage where you don't know the exact slot would require a large switch statement. I'd prefer to remove the methods for each slot in favor of the general #getEquipment, but that's more verbose.